From c14160ba5e5791f1544c7b5ceaafbb3bdb05836e Mon Sep 17 00:00:00 2001
From: Van Do <van.do.xw@renesas.com>
Date: Wed, 17 Oct 2018 14:59:24 +0700
Subject: [PATCH 036/909] soc: renesas: rcar-sysc: Include 3DG as always-on
 power domains

commit 0e128999c0ad8161cf43f0a9bd2e3ca4fe165e27 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

As in commit (soc: renesas: rcar-sysc: Improve power
domains always on implementation (1*)),  it shows that
Hardware needs to be controlled with the state transitions
do not conflict for each power domains.

After verifying again, the patterns of conflict are updated as below.

1. CA5x  vs CA5x : Exclusion control with HW
2. CA5x  vs 3DG  : Exclusion control with SW (Not implemented yet)
3. CA5x  vs Other: Exclusion control with SW (Not implemented yet)
4. 3DG   vs Other: Exclusion control with SW (already implemented)
5. Other vs Other: Exclusion control with SW (already implemented)

In that,
- 3. has been implemented in (1*).
- 2. has not been implemented and this patch will implement for it.

Note: On Salvator-X, this pattern was only detected
	when SW11-4 is OFF (debug ON mode.)

Signed-off-by: Van Do <van.do.xw@renesas.com>
Signed-off-by: Dien Pham <dien.pham.ry@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/renesas/rcar-sysc.c | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/drivers/soc/renesas/rcar-sysc.c b/drivers/soc/renesas/rcar-sysc.c
index 3d70a914328a..64c11374e521 100644
--- a/drivers/soc/renesas/rcar-sysc.c
+++ b/drivers/soc/renesas/rcar-sysc.c
@@ -69,19 +69,27 @@ const struct soc_device_attribute rcar_sysc_quirks_match[] __initconst = {
 		.soc_id = "r8a7795", .revision = "ES2.0",
 		.data = (void *)(BIT(R8A7795_PD_A3VP) | BIT(R8A7795_PD_CR7)
 			| BIT(R8A7795_PD_A3VC) | BIT(R8A7795_PD_A2VC0)
-			| BIT(R8A7795_PD_A2VC1) | BIT(R8A7795_PD_A3IR)),
+			| BIT(R8A7795_PD_A2VC1) | BIT(R8A7795_PD_A3IR)
+			| BIT(R8A7795_PD_3DG_A) | BIT(R8A7795_PD_3DG_B)
+			| BIT(R8A7795_PD_3DG_C) | BIT(R8A7795_PD_3DG_D)
+			| BIT(R8A7795_PD_3DG_E)),
 	},
 	{
 		.soc_id = "r8a7795", .revision = "ES1.*",
 		.data = (void *)(BIT(R8A7795_PD_A3VP) | BIT(R8A7795_PD_CR7)
 			| BIT(R8A7795_PD_A3VC) | BIT(R8A7795_PD_A2VC0)
-			| BIT(R8A7795_PD_A2VC1) | BIT(R8A7795_PD_A3IR)),
+			| BIT(R8A7795_PD_A2VC1) | BIT(R8A7795_PD_A3IR)
+			| BIT(R8A7795_PD_3DG_A) | BIT(R8A7795_PD_3DG_B)
+			| BIT(R8A7795_PD_3DG_C) | BIT(R8A7795_PD_3DG_D)
+			| BIT(R8A7795_PD_3DG_E)),
+
 	},
 	{
 		.soc_id = "r8a7796", .revision = "ES1.*",
 		.data = (void *)(BIT(R8A7796_PD_CR7) | BIT(R8A7796_PD_A3VC)
 			| BIT(R8A7796_PD_A2VC0) | BIT(R8A7796_PD_A2VC1)
-			| BIT(R8A7796_PD_A3IR)),
+			| BIT(R8A7796_PD_A3IR) | BIT(R8A7796_PD_3DG_A)
+			| BIT(R8A7796_PD_3DG_B)),
 	},
 	{
 		.soc_id = "r8a77965", .revision = "ES1.0",
-- 
2.17.1

